Godeps/_workspace/src/k8s.io/kubernetes/pkg/apiserver/apiserver.go

```diff
@@ -177,7 +177,11 @@ func serviceErrorHandler(requestResolver *APIRequestInfoResolver, apiVersions []
 		}
 	}
 
-	errorJSON(apierrors.NewGenericServerResponse(serviceErr.Code, "", "", "", "", 0, false), codec, response.ResponseWriter)
+	resourceOrKind := requestInfo.Kind
+	if len(resourceOrKind) == 0 {
+		resourceOrKind = requestInfo.Resource
```

Comment:
Ambiguous messages are ok for now if we're not 100% sure what we want.
Make sure we have a follow up and I'm ok with that.

On Thu, Aug 13, 2015 at 11:01 AM, Paul Weil <notifications@github.com>
wrote:

> In Godeps/_workspace/src/k8s.io/kubernetes/pkg/apiserver/apiserver.go
> <https://github.com/openshift/origin/pull/4115#discussion_r36984424>:
>
> > @@ -177,7 +177,11 @@ func serviceErrorHandler(requestResolver *APIRequestInfoResolver, apiVersions []
> >  		}
> >  	}
> >
> > -	errorJSON(apierrors.NewGenericServerResponse(serviceErr.Code, "", "", "", "", 0, false), codec, response.ResponseWriter)
> > +	resourceOrKind := requestInfo.Kind
> > +	if len(resourceOrKind) == 0 {
> > +		resourceOrKind = requestInfo.Resource
>
> So...here's my opinion: in order to not get disabling the features and
> this larger info handling too coupled I could remove my changes here and
> just leave the "don't swallow error messages" upstream change. We'd be left
> with the ambiguous messages we already have. WDYT?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4115/files#r36984424>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift